### PR TITLE
Feature/upload profile image 175846487

### PR DESCRIPTION
**Pull request by JelenaM-Kolosek**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,11 +1,40 @@
 class User < ApplicationRecord
+  has_attached_file :image, default_url: ":style/avatar.png", :styles => { :medium => "300x300>", :thumb => "100x100>" }
+  validates_attachment_content_type :image, :content_type => ["image/jpg", "image/jpeg", "image/png"]
+
+  #validates :slug, uniqueness: true
+
+  def create_slug
+    #index = self.full_name.index(' ')
+   #self.full_name.downcase.strip.gsub(' ', '-').gsub(/[^\w-]/,'')[index+2..-1]
+
```
Comment (tamara-bol) [inline app/models/user.rb]: delete what is not needed, and correct identation in general

File: app/models/user.rb [hunk 2]
```diff
@@ -1,11 +1,40 @@
...
+  
+  end
+
+  def update_slug
+    update_attributes slug: assign_slug
+  end
+  private
+  def assign_slug
+    self.slug = create_slug
+  end
```
Comment (tamara-bol) [inline app/models/user.rb]: what is used here? create slug is an empty method

File: app/policies/users_policy.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class UsersPolicy < ApplicationPolicy
+      def update?
+        user.present?
```
Comment (tamara-bol) [inline app/policies/users_policy.rb]: can non admin user update other users?

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,10 @@
 Rails.application.routes.draw do
   root to: 'pages#home'
+  get "users/new_avatar" => 'users#new_avatar', :as => :new_avatar
```
Comment (tamara-bol) [inline config/routes.rb]: include this route withing users route, for example:

```
resources :users do
  member do
    get 'device_connect_edit'
    post 'device_connect_update'
  end
end
```

**Review by tamara-bol — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (JelenaM-Kolosek) [reply app/models/user.rb]: That is a part of profile overview branch. I changed it and it will be good when I merge them.

**Pull request merged**